File: sample-app-java+push-notification/src/main/java/com/cometchat/sampleapp/java/fcm/ui/adapters/CallDetailsTabHistoryAdapter.java
package com.cometchat.sampleapp.java.fcm.ui.adapters;

import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.content.res.AppCompatResources;
import androidx.recyclerview.widget.RecyclerView;

import com.cometchat.calls.constants.CometChatCallsConstants;
import com.cometchat.calls.model.CallLog;
import com.cometchat.calls.model.CallUser;
import com.cometchat.chatuikit.CometChatTheme;
import com.cometchat.chatuikit.calls.utils.CallUtils;
import com.cometchat.chatuikit.shared.resources.utils.Utils;
import com.cometchat.sampleapp.java.fcm.R;
import com.cometchat.sampleapp.java.fcm.databinding.CallDetailsHistoryItemsBinding;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class CallDetailsTabHistoryAdapter extends RecyclerView.Adapter<CallDetailsTabHistoryAdapter.MyViewHolder> {
    private final Context context;
    private List<CallLog> callLogs;

    public CallDetailsTabHistoryAdapter(Context context) {
        this.context = context;
        callLogs = new ArrayList<>();
    }

    public void setCallLogs(@Nullable List<CallLog> list) {
        if (list != null) {
            this.callLogs = list;
            notifyDataSetChanged();
        }
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        CallDetailsHistoryItemsBinding binding = CallDetailsHistoryItemsBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new MyViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        CallLog callLog = callLogs.get(position);
        boolean isLoggedInUser = CallUtils.isLoggedInUser((CallUser) callLog.getInitiator());
        boolean isMissedOrUnanswered = callLog.getStatus().equals(CometChatCallsConstants.CALL_STATUS_UNANSWERED) || callLog.getStatus().equals(
            CometChatCallsConstants.CALL_STATUS_MISSED);
        holder.binding.tvInfoDate.setDateText(Utils.callLogsTimeStamp(callLog.getInitiatedAt(), null));
        if (callLog.getType().equals(CometChatCallsConstants.CALL_TYPE_AUDIO) || callLog.getType()
                                                                                        .equals(CometChatCallsConstants.CALL_TYPE_VIDEO) || callLog
            .getType()
            .equals(CometChatCallsConstants.CALL_TYPE_AUDIO_VIDEO)) {
            if (isLoggedInUser) {
                holder.binding.tvInfoTitle.setText(R.string.app_call_outgoing);
                holder.binding.tvInfoTitle.setTextAppearance(CometChatTheme.getTextAppearanceHeading4Medium(context));
                holder.binding.tvInfoTitle.setTextColor(CometChatTheme.getTextColorPrimary(context));
                setupCallIcon(
                    holder.binding.ivInfoIcon,
                    AppCompatResources.getDrawable(context, com.cometchat.chatuikit.R.drawable.cometchat_ic_outgoing_call),
                    CometChatTheme.getSuccessColor(context)
                );
            } else if (isMissedOrUnanswered) {
                holder.binding.tvInfoTitle.setText(R.string.app_call_missed);
                holder.binding.tvInfoTitle.setTextAppearance(CometChatTheme.getTextAppearanceHeading4Medium(context));
                holder.binding.tvInfoTitle.setTextColor(CometChatTheme.getErrorColor(context));
                setupCallIcon(
                    holder.binding.ivInfoIcon,
                    AppCompatResources.getDrawable(context, com.cometchat.chatuikit.R.drawable.cometchat_ic_missed_call),
                    CometChatTheme.getErrorColor(context)
                );
            } else {
                holder.binding.tvInfoTitle.setText(R.string.app_call_incoming);
                holder.binding.tvInfoTitle.setTextAppearance(CometChatTheme.getTextAppearanceHeading4Medium(context));
                holder.binding.tvInfoTitle.setTextColor(CometChatTheme.getTextColorPrimary(context));
                setupCallIcon(
                    holder.binding.ivInfoIcon,
                    AppCompatResources.getDrawable(context, com.cometchat.chatuikit.R.drawable.cometchat_ic_incoming_call),
                    CometChatTheme.getSuccessColor(context)
                );
            }
        }

        holder.binding.tvInfoCallDuration.setText(getCallDuration(callLog.getTotalDurationInMinutes()));
    }

    private void setupCallIcon(ImageView imageView, Drawable icon, @ColorInt int iconTint) {
        imageView.setBackground(icon);
        imageView.setBackgroundTintList(ColorStateList.valueOf(iconTint));
    }

    private String getCallDuration(double totalDurationInMinutes) {
        int minutes, seconds;
        minutes = (int) totalDurationInMinutes;
        seconds = (int) ((totalDurationInMinutes - minutes) * 60);
        return String.format(Locale.US, "%dm %ds", minutes, seconds);
    }

    @Override
    public int getItemCount() {
        return callLogs.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        CallDetailsHistoryItemsBinding binding;

        public MyViewHolder(@NonNull CallDetailsHistoryItemsBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: sample-app-kotlin+push-notification/src/main/java/com/cometchat/sampleapp/kotlin/fcm/ui/adapters/CallDetailsTabHistoryAdapter.kt
package com.cometchat.sampleapp.kotlin.fcm.ui.adapters

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import androidx.annotation.ColorInt
import androidx.appcompat.content.res.AppCompatResources
import androidx.recyclerview.widget.RecyclerView
import com.cometchat.calls.constants.CometChatCallsConstants
import com.cometchat.calls.model.CallLog
import com.cometchat.calls.model.CallUser
import com.cometchat.chatuikit.CometChatTheme
import com.cometchat.chatuikit.calls.utils.CallUtils
import com.cometchat.chatuikit.shared.resources.utils.Utils
import com.cometchat.sampleapp.kotlin.fcm.R
import com.cometchat.sampleapp.kotlin.fcm.databinding.CallDetailsHistoryItemsBinding
import java.util.Locale

class CallDetailsTabHistoryAdapter(private val context: Context) : RecyclerView.Adapter<CallDetailsTabHistoryAdapter.MyViewHolder>() {
    private var callLogs: List<CallLog>

    init {
        callLogs = ArrayList()
    }

    fun setCallLogs(list: List<CallLog>) {
        this.callLogs = list
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): MyViewHolder {
        val binding = CallDetailsHistoryItemsBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return MyViewHolder(binding)
    }

    override fun onBindViewHolder(
        holder: MyViewHolder,
        position: Int
    ) {
        val callLog = callLogs[position]
        val isLoggedInUser = CallUtils.isLoggedInUser(callLog.initiator as CallUser)
        val isMissedOrUnanswered =
            callLog.status == CometChatCallsConstants.CALL_STATUS_UNANSWERED || callLog.status == CometChatCallsConstants.CALL_STATUS_MISSED
        holder.binding.tvInfoDate.dateText = Utils.callLogsTimeStamp(
            callLog.initiatedAt, null
        )
        if (callLog.type == CometChatCallsConstants.CALL_TYPE_AUDIO || callLog.type == CometChatCallsConstants.CALL_TYPE_VIDEO || callLog.type == CometChatCallsConstants.CALL_TYPE_AUDIO_VIDEO) {
            if (isLoggedInUser) {
                holder.binding.tvInfoTitle.setText(R.string.app_call_outgoing)
                holder.binding.tvInfoTitle.setTextAppearance(
                    CometChatTheme.getTextAppearanceHeading4Medium(
                        context
                    )
                )
                holder.binding.tvInfoTitle.setTextColor(CometChatTheme.getTextColorPrimary(context))
                setupCallIcon(
                    holder.binding.ivInfoIcon, AppCompatResources.getDrawable(
                        context, com.cometchat.chatuikit.R.drawable.cometchat_ic_outgoing_call
                    ), CometChatTheme.getSuccessColor(
                        context
                    )
                )
            } else if (isMissedOrUnanswered) {
                holder.binding.tvInfoTitle.setText(R.string.app_call_missed)
                holder.binding.tvInfoTitle.setTextAppearance(
                    CometChatTheme.getTextAppearanceHeading4Medium(
                        context
                    )
                )
                holder.binding.tvInfoTitle.setTextColor(CometChatTheme.getErrorColor(context))
                setupCallIcon(
                    holder.binding.ivInfoIcon, AppCompatResources.getDrawable(
                        context, com.cometchat.chatuikit.R.drawable.cometchat_ic_missed_call
                    ), CometChatTheme.getErrorColor(
                        context
                    )
                )
            } else {
                holder.binding.tvInfoTitle.setText(R.string.app_call_incoming)
                holder.binding.tvInfoTitle.setTextAppearance(
                    CometChatTheme.getTextAppearanceHeading4Medium(
                        context
                    )
                )
                holder.binding.tvInfoTitle.setTextColor(CometChatTheme.getTextColorPrimary(context))
                setupCallIcon(
                    holder.binding.ivInfoIcon, AppCompatResources.getDrawable(
                        context, com.cometchat.chatuikit.R.drawable.cometchat_ic_incoming_call
                    ), CometChatTheme.getSuccessColor(
                        context
                    )
                )
            }
        }

        holder.binding.tvInfoCallDuration.text = getCallDuration(callLog.totalDurationInMinutes)
    }

    private fun setupCallIcon(
        imageView: ImageView,
        icon: Drawable?,
        @ColorInt iconTint: Int
    ) {
        imageView.background = icon
        imageView.backgroundTintList = ColorStateList.valueOf(iconTint)
    }

    private fun getCallDuration(totalDurationInMinutes: Double): String {
        val seconds: Int
        val minutes = totalDurationInMinutes.toInt()
        seconds = ((totalDurationInMinutes - minutes) * 60).toInt()
        return String.format(Locale.US, "%dm %ds", minutes, seconds)
    }

    override fun getItemCount(): Int {
        return callLogs.size
    }

    class MyViewHolder(var binding: CallDetailsHistoryItemsBinding) : RecyclerView.ViewHolder(binding.root)
}
